packages/opencensus-web-instrumentation-zone/src/interaction-tracker.ts

```diff
@@ -186,6 +199,46 @@ export class InteractionTracker {
     });
   }
 
+  private interceptXHRTasks(task: AsyncTask) {
+    if (isTrackedTask(task) && task.target instanceof XMLHttpRequest) {
+      const rootSpan: RootSpan = task.zone.get('data').rootSpan;
+      const xhr = task.target as XHRWithUrl;
+      if (xhr.readyState === XMLHttpRequest.OPENED) {
+        const childSpan = rootSpan.startChildSpan({
+          name: parseUrl(xhr.__zone_symbol__xhrURL).pathname,
+          kind: SpanKind.CLIENT,
+        });
+        // Associate the XHR to the child span ID so it allows to
+        // find the correct span to end it when the request is DONE.
+        this.xhrTasks[childSpan.id] = xhr;
+        if (traceOriginMatchesOrSameOrigin()) {
+          xhr.setRequestHeader(
+            'traceparent',
+            spanContextToTraceParent({
+              traceId: rootSpan.traceId,
+              spanId: childSpan.id,
+            })
+          );
+        }
+      } else if (xhr.readyState === XMLHttpRequest.DONE) {
+        const spanId =
+          Object.keys(this.xhrTasks).find(
+            spanId => this.xhrTasks[spanId] === xhr
+          ) || '';
+        const childSpan = rootSpan.spans.find(span => span.id === spanId);
+        if (childSpan) {
+          childSpan.addAttribute(
+            ATTRIBUTE_HTTP_STATUS_CODE,
+            xhr.status.toString()
```

Comment:
Why do you use `toString` here? I think we can actually set the status as a number

Reply:
As the type `number` in Typescript is a double-precision 64-bit, if we send just the number, when the JSON object is created for this span, the type value will be `doublevalue` then the status code is not shown, but if a string is sent rather than a number it work fine.

Reply:
Interesting, I think that's probably a bug in how it gets exported to Stackdriver, but for now I'm OK with making it a string. Could you add a TODO that we should investigate changing this to a number once we know that it can be exported correctly?